types/iobroker/index.d.ts

```diff
@@ -266,6 +266,28 @@ declare global {
             recursive?: boolean;
         }
 
+        interface ExtendObjectOptionsPreserve {
+            [prop: string]: ExtendObjectOptionsPreserve | boolean | string[];
+        }
+
+        interface ExtendObjectOptions {
+            /** Which properties of the original object should be preserved */
+            preserve?: ExtendObjectOptionsPreserve;
```

Comment:
We have `user?: string` too, but I think it's also not documented for the other object methods currently

Reply:
However, additional should be allowed if we say we are not sure yet what is allowed